### DISCUSSION TITLE
[Bug]: Tooltip/title Not Displaying While Tag Icon is Moving on Fabric.js Canvas

### DISCUSSION THREAD

**Discussion by tvprasad** [fabricjs/fabric.js#9999] [General]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.1

### In What environments are you experiencing the problem?

Firefox, Chrome, Microsoft Edge

### Node Version (if applicable)

None

### Link To Reproduction

Next template >= v6

### Steps To Reproduce

1. The tooltip associated with a tag icon does not display while the icon is being moved on the Fabric.js canvas. The tooltip is only visible when the tag icon is stationary.

Steps to Reproduce:

1. Add a tag icon to the Fabric.js canvas.
2. Attach a tooltip to the tag icon.
3. Move the tag icon around the canvas.

Video recording is attached.

https://github.com/fabricjs/fabric.js/assets/915494/097ef091-d5c0-43bf-b0c3-5419edc6c7f0

> 		
Find the Tag Object
Update the x,y when Found
Create a Tag Image, Radius Circle Since Tag Not Found
Create a Tooltip Div and Attach it to Tag Mouse Events

`		
        var foundTag = false;
        const objectId = 'Tag_${tagMac}_';
        const tagRadiusId = 'Tag_Radius_${tagMac}_${tagNumber}_';
        xPixels = calX(x);
        yPixels = calY(y);
		const t = `_${tagNumber}_`;
		//Find the Tag Object
        const imageObject = canvas.getObjects().find(x => x.id && x.id.includes(t) && x.type === 'image');
        let _title = `${tagNumber}_${locatedDate}_${caller}`
        let _fillColor = getColor(setAlarm);
		//Update the x,y when found
        if (imageObject) {
            foundTag = true;
            if (isRemoveLKTL) { canvas.remove(imageObject); } else {
                imageObject.setSrc(imgUrl, function () {
                    imageObject.set({ left: xPixels, top: yPixels, originX: 'center', originY: 'center', fill: _fillColor, scaleX: 0.1, scaleY: 0.1 });
					title: _title });
				});
            }
        }
        const tagRadius = canvas.getObjects().find(obj => obj.id && obj.id.startsWith(tagRadiusId) && obj.type === 'circle');
        if (tagRadius) {
            if (isRemoveLKTL) { canvas.remove(tagRadius); } else {
                tagRadius.set({ left: xPixels, top: yPixels, originX: 'center', originY: 'center', title: _title, radius: 150, dirty: false });
            }
        }
        if (foundTag || isRemoveLKTL) {
			canvas.calcOffset();  canvas.renderAll();
			return;
        }
        var _id = `Tag_${tagMac}_${tagNumber}_image`;
        const transparentCircle = new fabric.Circle({
            left: xPixels,
            top: yPixels,
            radius: radiusSize,
            fill: _fillColor,
            originX: 'center',
            originY: 'center',
            selectable: false,
            evented: false,
            title: _title,
            id: tagRadiusId
        });
        fabric.Image.fromURL(imgUrl, function(img) {
            img.set({ id: _id, left: xPixels, top: yPixels, originX: 'center', originY: 'center',scaleX: 0.1, scaleY: 0.1, title: _title });
			img.hoverCursor = 'pointer';

            // Create a tooltip
            var tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            tooltip.style.position = 'absolute';
            tooltip.style.display = 'none';
            tooltip.style.padding = '5px';
            //tooltip.style.background = '#00000000';
            tooltip.style.color = 'white';
            tooltip.style.borderRadius = '3px';
            document.body.appendChild(tooltip);

            img.on('mouseover', function (e) {
                if (e.target && e.target.id) {
                if (e.target.id.startsWith('Tag_')) {
                     let t = e.target.title ? e.target.title : 'No Information';
                        tooltip.innerText = t;
                        tooltip.style.display = 'block';
                       
                 }
                    canvas.renderAll();
                }
            });

            img.on('mousemove', function (e) {
               transparentCircle.set({ left: img.left, top: img.top });
               if (tooltip.style.display === 'block') {
                    tooltip.style.left = (e.e.clientX + 10) + 'px';
                    tooltip.style.top = (e.e.clientY + 10) + 'px';
                }               
                canvas.renderAll();
            });

            img.on('mouseout', function (e) {
                tooltip.style.display = 'none';
                canvas.renderAll();
            });
            // Add circle, icon, and tooltip to the canvas
			canvas.add(img).renderAll();
            canvas.add(transparentCircle).renderAll();        
        });
        canvas.renderAll();
		
		
		`

### Expected Behavior

The tooltip should be visible when focused on the tag icon during movement.

### Actual Behavior

The tooltip vanishes when the tag icon is moving, but it is visible both before the movement and after the tag icon is created.

### Error Message & Stack Trace

```bash
None
```

**Comment by zhe-he** [comment]
Since the code is not available, I wrote a simple demo, and it seems to be working fine.

https://github.com/fabricjs/fabric.js/assets/11306583/1e648094-060e-442b-96b6-9c692c49ae72

[demo](https://codesandbox.io/p/devbox/fabric-icon-mouseover-936sqy?file=%2Fsrc%2FApp.tsx%3A36%2C12)

**Comment by tvprasad** [comment]
@zhe-he,

Thank you for the sample demo. I am using fabricjs. My issue is that the tooltip disappears when the image is moved programmatically.

**Comment by zhe-he** [comment]
@tvprasad 
You can click on the demo link below to see the details. You will find that the rectangle in the video is an image... Or, you can present the bug in the way I do. You can delete or add anything in the file of my demo link to reproduce the bug you encountered.
